### ISSUE TITLE
Feature Request: "Intelligent" auto increment

### ISSUE THREAD

**Issue by faxesystem** [snipe/snipe-it#4577]
#### Expected Behavior (or desired behavior if a feature request)

If auto increment is used it should be a bit more intelligent and not simply counting +1. For example
You have Asset 1, 2, 3 and 4. The next auto increment would be 5. In the meantime Asset 2 gets deleted. Instead of auto increment to 5 it should look for the lowest available Asset Tag and choose 2 as the auto asset tag.
Also it should just look for the lowest number after the set auto increment number. So lets say auto increment should start at 200 than it will not choose a free asset tag below 200.

TLDR: auto asset tags should look for the lowest not used asset tag before auto increment.

-----

#### Actual Behavior

Auto increment is just doing +1

-----

**Comment by cslucian** [comment]
I think that auto increment it meant to be unique so using a deleted value won't be "unique".

I have not tested but I think if you delete "asset 2" you can manually modify "asset 5" to "asset 2".

Best of luck

**Comment by faxesystem** [comment]
@cslucian this is possible. But for example in our company there is one team member responsible for the leaver process meaning he deletes all user accounts, collects the inventory and deletes for example a screen if it is to old to be given to another person. Now another team member responsible for the purchase process adds new assets to snipe but he doesn't know that for example asset tag 1234 is free since it has been deleted. 

Using the intelligent auto increment would prevent the "swiss cheese" scenario with asset tags.

**Comment by tbradsha** [comment]
Is there ever any reason to need to reuse old asset tags, though? What's the advantage of filling in the gaps vs. moving on to a new, never-used asset number? It might look cleaner, but can be a nightmare when dealing with references (say, hand-written notes) that might be referring to the old asset with tag ID 1234 instead of the new asset with tag ID 1234.

I don't reuse asset tags, as that allows me to roughly gauge when something was assigned simply by knowing the number, since they're assigned ascending and thereby chronologically.
